include/xgboost/gbm.h

```diff
@@ -70,6 +70,10 @@ class GradientBooster {
   virtual void DoBoost(DMatrix* p_fmat,
                        std::vector<bst_gpair>* in_gpair,
                        ObjFunction* obj = nullptr) = 0;
+  virtual void DoBoost(DMatrix* p_fmat,
```

Comment:
Is it possible to only have a single function signature here? i.e. can we use the dhvec for everything?

Reply:
Having a single function signature here (and in other headers) would require modifying a large number of additional files and tests. For example, `DoBoost()` is also defined by `GBLinear`, which is not otherwise modified in this pull request.

I'd prefer to make such a change in a separate pull request, if possible.

Reply:
This could be a possibility. My main concern here is that Nvidia gets other priorities internally and I am left to clean it up :)